contentcuration/contentcuration/db/models/manager.py

```diff
@@ -74,54 +122,55 @@ def _move_child_to_new_tree(self, node, target, position):
-            return super(CustomContentNodeTreeManager, self)._move_node(
-                node, target, position, save, refresh_target
-            )
-
-    def insert_node(
-        self,
-        node,
-        target,
-        position="last-child",
-        save=False,
-        allow_existing_pk=False,
-        refresh_target=True,
-    ):
-        with self._update_changes(node, save):
-            if save:
-                with self.lock_mptt(target.tree_id):
-                    return super(CustomContentNodeTreeManager, self).insert_node(
-                        node, target, position, save, allow_existing_pk, refresh_target
-                    )
-            return super(CustomContentNodeTreeManager, self).insert_node(
-                node, target, position, save, allow_existing_pk, refresh_target
+    def _mptt_refresh(self, *nodes):
+        ids = [node.id for node in nodes if node.id]
+        if not ids:
+            return
+        opts = self.model._mptt_meta
+        values_lookup = {
+            c["id"]: c
+            for c in self.filter(id__in=ids).values(
+                "id",
+                opts.left_attr,
+                opts.right_attr,
+                opts.level_attr,
+                opts.tree_id_attr,
             )
-
-    def move_node(self, node, target, position="first-child"):
+        }
+        for node in nodes:
+            if node.id:
+                values = values_lookup[node.id]
+                for k, v in values.items():
+                    setattr(node, k, v)
+
+    def move_node(self, node, target, position="last-child"):
+        """
+        Vendored from mptt - by default mptt moves (with a save) then saves again
+        This is updated to just do the move, and no additional save.
+
+        Moves ``node`` relative to a given ``target`` node as specified
+        by ``position`` (when appropriate), by examining both nodes and
+        calling the appropriate method to perform the move.
+        A ``target`` of ``None`` indicates that ``node`` should be
+        turned into a root node.
+        Valid values for ``position`` are ``'first-child'``,
+        ``'last-child'``, ``'left'`` or ``'right'``.
+        ``node`` will be modified to reflect its new tree state in the
+        database.
+        This method explicitly checks for ``node`` being made a sibling
+        of a root node, as this is a special case due to our use of tree
+        ids to order root nodes.
+        NOTE: This is a low-level method; it does NOT respect
+        ``MPTTMeta.order_insertion_by``.  In most cases you should just
+        move the node yourself by setting node.parent.
+        """
         with self.lock_mptt(node.tree_id, target.tree_id):
-            super(CustomContentNodeTreeManager, self).move_node(node, target, position)
+            self._mptt_refresh(node, target)
+            self._move_node(node, target, position=position)
+            node.save(skip_lock=True)
```

Comment:
In `TreeManager`, the `_move_node` function appears to default the `save` argument to `True`, so aren't we still calling save twice here?

Reply:
I thought that at first - but the `save` argument to `_move_node` only does something if we are inside `delay_mptt_updates`: https://github.com/django-mptt/django-mptt/blob/master/mptt/managers.py#L552

Otherwise it calls methods that do not call a save - https://github.com/django-mptt/django-mptt/blob/master/mptt/managers.py#L562

Reply:
(hence my misleading code comment above)

Reply:
Ah, gotcha, the other codepaths trigger the custom SQL updates in mptt. My brain is trained to see `with` blocks for mptt code as setting a delay / disable updates flag!